components/raftstore/src/store/fsm/apply.rs

```diff
@@ -945,16 +952,38 @@ where
         &mut self,
         apply_ctx: &mut ApplyContext<EK, W>,
         entry: &Entry,
+        entry_type: EntryType,
     ) -> ApplyResult<EK::Snapshot> {
         // Although conf change can't yield in normal case, it is convenient to
         // simulate yield before applying a conf change log.
         fail_point!("yield_apply_conf_change_3", self.id() == 3, |_| {
             ApplyResult::Yield
         });
-        let index = entry.get_index();
-        let term = entry.get_term();
-        let conf_change: ConfChange = util::parse_data_at(entry.get_data(), index, &self.tag);
-        let cmd = util::parse_data_at(conf_change.get_context(), index, &self.tag);
+        let (index, term) = (entry.get_index(), entry.get_term());
+        let (conf_change, cmd) = match entry_type {
+            EntryType::EntryConfChange => {
+                let conf_change: ConfChange =
+                    util::parse_data_at(entry.get_data(), index, &self.tag);
+                let mut cmd: RaftCmdRequest =
+                    util::parse_data_at(conf_change.get_context(), index, &self.tag);
+                {
+                    // Covert ChangePeerRequest to ChangePeerV2Request
```

Comment:
Why not convert it before actually applying?

Reply:
Do you mean convert it before proposing?

Reply:
I mean move it to `exec_conf_change`.